math-scala/src/main/scala/org/apache/mahout/math/algorithms/transformer/MeanCenter.scala

```
@@ -0,0 +1,93 @@
+  * software distributed under the License is distributed on an
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  * KIND, either express or implied. See the License for the
+  * specific language governing permissions and limitations
+  * under the License.
+  */
+
+package org.apache.mahout.math.algorithms.transformer
+
+import collection._
+import JavaConversions._
+import org.apache.mahout.math.drm._
+import org.apache.mahout.math.drm.RLikeDrmOps._
+import org.apache.mahout.math.{Matrix, Vector}
+import org.apache.mahout.math.scalabindings.RLikeOps._
+import org.apache.mahout.math.scalabindings.MahoutCollections._
+import org.apache.mahout.math.{Vector => MahoutVector}
+
+import scala.reflect.ClassTag
+
+class MeanCenter extends Transformer {
+
+  var summary = ""
+  var colMeansV: MahoutVector = _
+
+  /**
+    * Optionally set the centers of each column to some value other than Zero
+    * @param centers A vector of length equal to the `input` in the fit method specifying the
+    *                centers to set each column to.
+    */
+  def setCenter(centers: MahoutVector) = {
+    colMeansV = colMeansV - centers
+  }
+
+  /**
+    * Centers Columns at zero
+    * @param input
+    */
+  def fit[K](input: DrmLike[K]) = {
+    colMeansV = input.colMeans
+    val colMeansA = colMeansV.toArray
+    //summary = (0 until colMeansA.length).map(i => s"Column ${i} mean: ${colMeansA(i)}").mkString(", ")
+  }
+
+  def transform[K: ClassTag](input: DrmLike[K]): DrmLike[K] = {
+
+    if (!isFit) {
+      //throw an error
+    }
+
+    implicit val ctx = input.context
+    val bcastV = drmBroadcast(colMeansV)
+
+    val output = input.mapBlock(input.ncol) {
+      case (keys, block) =>
+        val copy: Matrix = block.cloned
+        copy.foreach(row => row -= bcastV.value)
+        (keys, copy)
+    }
+    output
+  }
+
+  def invTransform[K: ClassTag](input: DrmLike[K]): DrmLike[K] = {
+
+    if (!isFit) {
+      //throw an error
```

Same qestion here as before- do you want to throw an error here?